app/models/my_module.rb

```diff
@@ -10,6 +10,9 @@ class MyModule < ApplicationRecord
   include PermissionCheckableModel
   include Assignable
 
+  ID_PREFIX = 'TA'
+  include PrefixedIdModel
```

Comment:
The SEARCHABLE_ATTRIBUTES constant needs to be moved below this, and `PREFIXED_ID_SQL` added to it. This is necessary for global search to work properly with prefixed IDs. For reference check `app/models/experiment.rb`.